File: implementation/scipamato/scipamato-core-web/src/test/java/ch/difty/scipamato/core/web/security/ScipamatoAuthenticatedWebSessionConfigTest.java
package ch.difty.scipamato.core.web.security;

import static org.assertj.core.api.Assertions.assertThat;

import com.giffing.wicket.spring.boot.starter.configuration.extensions.external.spring.security.SecureWebSession;
import org.junit.Test;
import org.springframework.beans.factory.annotation.Autowired;

import ch.difty.scipamato.core.web.WicketTest;
import ch.difty.scipamato.core.web.authentication.ScipamatoAuthenticatedWebSessionConfig;

public class ScipamatoAuthenticatedWebSessionConfigTest extends WicketTest {

    @Autowired
    private ScipamatoAuthenticatedWebSessionConfig config;

    @Test
    public void canWire() {
        assertThat(config).isNotNull();
    }

    @Test
    public void providesSecureWebSession() {
        assertThat(config.getAuthenticatedWebSessionClass()).isNotNull();
        assertThat(config
            .getAuthenticatedWebSessionClass()
            .getName()).isEqualTo(SecureWebSession.class.getName());
    }
}

File: core/core-web/src/test/kotlin/ch/difty/scipamato/core/web/security/ScipamatoAuthenticatedWebSessionConfigTest.kt
package ch.difty.scipamato.core.web.security

import ch.difty.scipamato.core.web.WicketTest
import ch.difty.scipamato.core.web.authentication.ScipamatoAuthenticatedWebSessionConfig
import com.giffing.wicket.spring.boot.starter.configuration.extensions.external.spring.security.SecureWebSession
import org.amshove.kluent.shouldBeEqualTo
import org.amshove.kluent.shouldNotBeNull
import org.junit.jupiter.api.Test
import org.springframework.beans.factory.annotation.Autowired

internal class ScipamatoAuthenticatedWebSessionConfigTest : WicketTest() {

    @Autowired
    private lateinit var config: ScipamatoAuthenticatedWebSessionConfig

    @Test
    fun canWire() {
        config.shouldNotBeNull()
    }

    @Test
    fun providesSecureWebSession() {
        config.authenticatedWebSessionClass.shouldNotBeNull()
        config.authenticatedWebSessionClass.name shouldBeEqualTo SecureWebSession::class.java.name
    }
}
